ethcore/builtin/src/lib.rs

```diff
@@ -211,89 +227,128 @@ impl Builtin {
 
 	/// Whether the builtin is activated at the given block number.
 	pub fn is_active(&self, at: u64) -> bool {
-		at >= self.activate_at
+		self.pricer.keys().rev().any(|&other_at| at >= other_at)
 	}
 }
 
 impl From<ethjson::spec::Builtin> for Builtin {
 	fn from(b: ethjson::spec::Builtin) -> Self {
-		let pricer: Box<dyn Pricer> = match b.pricing {
-			ethjson::spec::Pricing::Blake2F { gas_per_round } => {
-				Box::new(gas_per_round)
+		// TODO(niklasad1): change this to `try_from` and propogate the error accordingly
+		let native = EthereumBuiltin::from_str(&b.name).unwrap();
+		let pricer = match b.pricing {
+			ethjson::spec::builtin::Pricing::Single(pricer) => {
+				// TODO(niklasad1): change chain specs and don't enable by default!?
+				// Seems more reasonable to do it the otherway around but would be a breaking change
+				map![b.activate_at.map_or(0, Into::into) => pricer.into()]
+			}
+
+			ethjson::spec::builtin::Pricing::Multi(pricer) => {
+				assert!(!pricer.is_empty(), "chainspec contains multiple builtins; but no values found");
```

Comment:
Can this be a `static_assert!`? Aren't we reading in the json file at compile time with `include_bytes!`?

Reply:
That would be cool, I will check tomorrow

Reply:
I don't think that's possible, rust doesn't have dependent types and you can't statically guarantee vec is not empty.

Reply:
well, can't you do in a build-script with a special deserialization function which denies empty vectors? 

However, we support loading custom chain specs at runtime so I think we can't guarantee that statically anyways,

Reply:
Maybe it shouldn't be a `Vec`? ;)

It doesn't matter here ofc, just curious. In theory it sounds like it should be possible to do static asserts on bytes from `include_bytes!`, but then again maybe it requires serde to run at compile time too...